### PR TITLE
Fix -dockerized execution #3137

### PR DESCRIPTION
**Pull request by Kibubu**
Fix dockerize execution https://github.com/nextflow-io/nextflow/issues/3137

Amazon's Corretto docker image is missing basic user management utilities the entry script relies on.

Added utilities to Dockerfile
Fixed execution of nextflow within docker Container

### The utilities adduser and addgroup are not installed inside the container image

Changed to useradd and groupadd added their dependencies to Dockerfile

### Fix could not create plugins folder

The permissions of /.nextflow were to restricted to

### Fix Typo and move comments to correct location

---

## Test script
```shell
#! /bin/bash

git clone  git@github.com:kibubu/nextflow.git fix-dockerize
cd fix-dockerize/docker || exit
git checkout Fix-dockerize
curl -s https://get.nextflow.io | bash
mkdir -p dist/linux/amd64
cp "$(which docker)" dist/linux/amd64
docker rmi nextflow/nextflow:22.04.5
docker build . -t nextflow/nextflow:22.04.5
mkdir tmp && cd tmp || exit
../nextflow -dockerize run nextflow-io/hello
sleep 5
which tree || exit
tree -a .
cd ..
```

[asciicast](https://asciinema.org/a/Nj8HZrXesn2hSSme8Xn0tYXll)

### CODE REVIEW

File: docker/entry.sh [hunk 1]
```diff
@@ -39,20 +39,27 @@
 # wrap cli args with single quote to avoid wildcard expansion
 cli=''; for x in "$@"; do cli+="'$x' "; done
 
-# the NXF_USRMAP hold the user ID in the host environment 
+# the NXF_USRMAP hold the user ID in the host environment
 if [[ "$NXF_USRMAP" ]]; then
-# create a `nextflow` user with the provided ID 
-# then change the docker socker ownership to `nextflow` user 
-addgroup docker
-adduser -u $NXF_USRMAP -G docker -s /bin/bash -D nextflow
```
Comment (pditommaso) [inline docker/entry.sh:47]: why `-D` has been removed?

**Comment by Kibubu** [comment]
I messed, totally forget that this PR was still open and deleted my fork, since I was to lazy to resolve merge conflicts.

Let's see what happens if I press reopen

...seriously embarrassed right now

File: docker/entry.sh [hunk 2]
```diff
@@ -39,20 +39,27 @@
...
-chown nextflow /var/run/docker.sock  
+# create a `nextflow` user with the provided ID
+
```
Comment (pditommaso) [inline docker/entry.sh:45]:
```suggestion
```

**Review by pditommaso — APPROVED** [review]

**Comment by pditommaso** [comment]
Thanks for this contribution

**Pull request merged**